src/peergos/server/tests/simulation/Simulator.java

```diff
@@ -301,6 +315,26 @@ private boolean grantPermission(String granter, String grantee, Path path, FileS
         FileSystem testFileSystem = fileSystems.getTestFileSystem(granter);
         FileSystem referenceFileSystem = fileSystems.getReferenceFileSystem(granter);
 
+        Set<String> testExistingWriters = new TreeSet<>(testFileSystem.getSharees(path, FileSystem.Permission.WRITE));
+        Set<String> refExistingWriters = new TreeSet<>(referenceFileSystem.getSharees(path, FileSystem.Permission.WRITE));
+        if (! testExistingWriters.toString().equals(refExistingWriters.toString())) {
```

Comment:
nit-pick: why convert to string when comparing?

Reply:
force of habit, fixed